core-services/src/main/java/org/zalando/nakadi/validation/ResourceValidationHelperService.java

```diff
@@ -0,0 +1,149 @@
+package org.zalando.nakadi.validation;
+
+import org.springframework.stereotype.Component;
+import org.zalando.nakadi.domain.ResourceAnnotations;
+import org.zalando.nakadi.domain.ResourceLabels;
+import org.zalando.nakadi.exceptions.runtime.InvalidResourceAnnotationException;
+import org.zalando.nakadi.exceptions.runtime.InvalidResourceLabelException;
+
+import java.util.Optional;
+import java.util.regex.Pattern;
+
+@Component
+public final class ResourceValidationHelperService {
+
+    private static final Pattern ALNUM_START_END_PATTERN = Pattern.compile("^[a-zA-Z0-9](.*[a-zA-Z0-9])?$");
+    private static final Pattern ALLOWED_CHARACTER_PATTERN = Pattern.compile("^[a-zA-Z0-9_.-]+$");
+    private static final String DNS_LABEL_FORMAT = "[a-z0-9]([-a-z0-9]*[a-z0-9])?";
+    private static final Pattern DNS_1123_SUBDOMAIN_PATTERN = Pattern
+            .compile("^" + DNS_LABEL_FORMAT + "(\\." + DNS_LABEL_FORMAT + ")*$");
+
+    private static final int MAX_KEY_NAME_LENGTH = 63;
+    private static final int MAX_PREFIX_LENGTH = 253;
+
+    static final int MAX_ANNOTATION_LENGTH = 1000;
+    static final int MAX_LABEL_LENGTH = 63;
+
+    private static final String ERROR_MESSAGE_TEMPLATE = "Error validating <%s:%s>; %s";
+    private static final String EMPTY_KEY_ERROR = "Key cannot be empty.";
+    private static final String MULTI_PREFIX_ERROR = "Key cannot have multiple prefixes.";
+    private static final String EMPTY_KEY_PREFIX_ERROR = "Key prefix cannot be empty.";
+    private static final String EMPTY_KEY_NAME_ERROR = "Key name cannot be empty.";
+    private static final String LONG_KEY_NAME_ERROR = "Key name cannot be more than 63 characters.";
+    private static final String START_END_KEY_NAME_ERROR = "Key name should start and end with [a-zA-Z0-9].";
+    private static final String START_END_LABEL_ERROR = "Label should start and end with [a-zA-Z0-9].";
+    private static final String KEY_NAME_CHARACTER_ERROR = "Key name should use only [a-zA-Z0-9] " +
+            "and `-`, '_', and `.`.";
+    private static final String LABEL_CHARACTER_ERROR = "Label should use only [a-zA-Z0-9] and `-`, '_', and `.`.";
+    private static final String LONGER_PREFIX_ERROR = "Key prefix cannot be more than 253 characters.";
+    private static final String INVALID_PREFIX_ERROR = "Key prefix must follow RFC 1123 DNS subdomain format. \n" +
+            "It must contain one or more segments separated by '.'(dot), and each segment must start and end with " +
+            "[a-z0-9], and may contain - (hyphen) and [a-z0-9] in between.";
+    private static final String PREFIX_SEPARATOR = "/";
+    private static final String LONG_ANNOTATION_ERROR = "Annotation cannot be more than " + MAX_ANNOTATION_LENGTH
+            + " characters long.";
+    private static final String LONG_LABEL_ERROR = "Label cannot be more than " + MAX_LABEL_LENGTH + "characters long.";
```

Comment:
```suggestion
    private static final String LONG_LABEL_ERROR = "Label cannot be more than " + MAX_LABEL_LENGTH + " characters long.";
```
a space is still missing

Reply:
:+1: